atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -495,6 +509,32 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         return transactionManager;
     }
 
+    @VisibleForTesting
+    static Optional<PersistentTimestampStore> constructPersistentTimestampStore(
+            AtlasDbConfig config,
+            PersistentStorageFactory persistentStorageFactory,
+            @Output List<AutoCloseable> closeables) {
+        return initializeCloseable(
+                config.persistentStorage().map(storageConfig -> persistentStorageFactory
+                        .constructPersistentTimestampStore((RocksDbPersistentStorageConfig) storageConfig)),
```

Comment:
I would `Preconditions.checkState` this for readability (instead of the class cast exception you'd get if we ship a bug here)

Reply:
Added.